src/modules/shorthand/add.rs

```diff
@@ -48,31 +48,47 @@ impl SyntaxModule<ParserMetadata> for ShorthandAdd {
 }
 
 impl TranslateModule for ShorthandAdd {
+    //noinspection DuplicatedCode
     fn translate(&self, meta: &mut TranslateMetadata) -> String {
-        let expr = self.is_ref
-            .then(|| self.expr.translate_eval(meta, true))
-            .unwrap_or_else(|| self.expr.translate(meta));
-        let name: String = match self.global_id {
-            Some(id) => format!("__{id}_{}", self.var),
-            None => if self.is_ref { format!("${{{}}}", self.var) } else { self.var.clone() }
+        let name = if let Some(id) = self.global_id {
+            format!("__{id}_{}", self.var)
+        } else if self.is_ref {
+            format!("${{{}}}", self.var)
+        } else {
+            self.var.clone()
         };
-        let stmt = match self.kind {
-            Type::Text => format!("{}+={}", name, expr),
-            Type::Array(_) => format!("{}+=({})", name, expr),
+        match self.kind {
+            Type::Text => {
+                if self.is_ref {
+                    let expr = self.expr.translate_eval(meta, true);
+                    format!("eval \"{name}+={expr}\"")
+                } else {
+                    let expr = self.expr.translate(meta);
+                    format!("{name}+={expr}")
+                }
+            }
+            Type::Array(_) => {
+                if self.is_ref {
```

Comment:
I unrolled some of the code which was previously shared across expression types in this `match` statement; this is necessary so we can append additional code to the `Array` case in the next PR.

Reply:
This should also make it slightly easier to implement @Ph0enixKM's proposed new translation layer.